### PR TITLE
Add pr description support

### PR DESCRIPTION
**Pull request by sharpyy**
### Description

Thanks for contributing this Pull Request. Make sure that you send in this Pull Request to the `master` branch of this repository, add a brief description, and tag the relevant issue(s) and PR(s) below.

- Relevant Issues :
Current PR have empty body.

- What Changed and Why? :

Added new feature for passing the PR documentation. Either as on the command line or passing a file path to a MarkDown file for example.
Updated the README with an example of simple transformation

- Type of change :
  - [X] New feature
  - [ ] Bug fix for existing feature
  - [ ] Code quality improvement
  - [X] Addition or Improvement of tests
  - [X] Addition or Improvement of documentation

### CODE REVIEW

File: gordian/gordian.py [hunk 1]
```diff
@@ -144,8 +156,15 @@ def apply_transformations(args, transformations, pr_created_callback=None):
...
 
+def get_pr_description(args):
+    if args.description_file is not None:
+        with open(args.description_file,'r') as fh:
+            return fh.read()
+    return args.description
+
 def transform(args, transformations, repositories, pr_created_callback):
     pull_request_urls = []
+    pr_description = get_pr_description(args)
```
Comment (AgarFu) [inline gordian/gordian.py:167]: I'd like to keep the interface of this transform as agnostic as possible, I think adding an extra `description` argument with '' as default value would be ideal.
Then in the apply_transformation I'd add the get_pr_description() call

```python
def apply_transformations(args, transformations, pr_created_callback=None):
    config = Config(args.config_file)	    config = Config(args.config_file)
    transform(args, transformations, config.get_data(), pr_created_callback=pr_created_callback, pr_description=get_pr_description(args))

def get_pr_description(args):
    if args.description_file is not None:
        with open(args.description_file,'r') as fh:
            return fh.read()
    return args.description

def transform(args, transformations, repositories, pr_created_callback, description=''):
    ...
```

File: gordian/gordian.py [hunk 2]
```diff
@@ -155,7 +174,9 @@ def transform(args, transformations, repositories, pr_created_callback):
             repo.bump_version(args.dry_run)
             if not args.dry_run:
                 try:
-                    pull_request = repo.create_pr(args.pr_message, '', args.target_branch, args.pr_labels)
+                    pull_request = repo.create_pr(args.pr_message, pr_description, args.target_branch, args.pr_labels)
+                    pull_request = repo._repo.create_pull(args.pr_message, pr_description, args.target_branch, repo.branch_name)
```
Comment (AgarFu) [inline gordian/gordian.py:178]: is this correct?

Comment (sharpyy) [reply gordian/gordian.py]: It was a rebase issue. It is fixed now.

File: gordian/gordian.py [hunk 3]
```diff
@@ -155,7 +174,8 @@ def transform(args, transformations, repositories, pr_created_callback):
             repo.bump_version(args.dry_run)
             if not args.dry_run:
                 try:
-                    pull_request = repo.create_pr(args.pr_message, '', args.target_branch, args.pr_labels)
+                    pull_request = repo.create_pr(args.pr_message, pr_description, args.target_branch, args.pr_labels)
+                    pull_request.set_labels(*args.pr_labels)
```
Comment (jnevelson) [inline gordian/gordian.py:178]: I think this was duplicated when rebasing possibly? The label is already being set here https://github.com/argoproj-labs/gordian/blob/master/gordian/repo.py#L204.

**Pull request merged**